src/kayenta/stages/kayentaStage/kayentaStage.controller.ts

```diff
@@ -0,0 +1,548 @@
+  };
+
+  private loadProviders = async (): Promise<string[]> => {
+    const providers = await AccountService.listProviders(this.$scope.application);
+    // TODO(dpeach): allow providers other than gce.
+    return this.providers = providers.filter(p => p === 'gce');
+  };
+
+  public handleProviderChange = async (): Promise<void> => {
+    set(this.stage, 'deployments.baseline.account', null);
+    set(this.stage, 'deployments.baseline.cluster', null);
+    this.accounts = await this.loadAccounts();
+    this.setClusterList();
+  };
+
+  public loadAccounts = (): Promise<IAccountDetails[]> => {
+    if (!has(this.stage, 'deployments.baseline.cloudProvider')) {
+      return null;
+    }
+
+    return Promise.resolve(AccountService.listAccounts(this.stage.deployments.baseline.cloudProvider));
+  };
+
+  private setClusterList = (): void => {
+    this.clusterList =
+      AppListExtractor.getClusters([this.$scope.application], sg =>
+        has(this.stage, 'deployments.baseline.account')
+          ? sg.account === this.stage.deployments.baseline.account
+          : true
+      );
+  };
+
+  public getRegion = (serverGroup: any): string => {
+    if (serverGroup.region) {
+      return serverGroup.region;
+    }
+    const availabilityZones = serverGroup.availabilityZones;
+    return availabilityZones
+      ? Object.keys(availabilityZones).length
+        ? Object.keys(availabilityZones)[0]
+        : 'n/a'
+      : 'n/a';
+  };
+
+  public getServerGroupName = (serverGroup: any): string => {
+    return NameUtils.getClusterName(
+      serverGroup.application,
+      serverGroup.stack,
+      serverGroup.freeFormDetails);
+  };
+
+  public addPair = async (): Promise<void> => {
+    this.stage.deployments.serverGroupPairs = this.stage.deployments.serverGroupPairs || [];
+    const provider =
+      await (
+        has(this.stage, 'deployments.baseline.cloudProvider')
+          ? Promise.resolve(this.stage.deployments.baseline.cloudProvider)
+          : Promise.resolve(this.providerSelectionService.selectProvider(this.$scope.application, 'serverGroup'))
+      );
+    this.stage.deployments.baseline.cloudProvider = provider;
+    const config = CloudProviderRegistry.getValue(provider, 'serverGroup');
+
+    const command: any =
+      await this.serverGroupCommandBuilder.buildNewServerGroupCommandForPipeline(
+        provider,
+        null,
+        null
+      );
+
+    command.viewState = {
+      ...command.viewState,
+      requiresTemplateSelection: true,
+      disableStrategySelection: true,
+      hideClusterNamePreview: true,
+      readOnlyFields: {
+        credentials: true,
+        region: true,
+        subnet: true,
+        useSourceCapacity: true
+      },
+      overrides: {
+        capacity: {
+          min: 1,
+          max: 1,
+          desired: 1,
+        },
+        useSourceCapacity: false,
+      },
+    };
+
+    delete command.strategy;
+    try {
+      const result = await this.$uibModal
```

Comment:
i can drop in and make this also work with the new amazon clone server group modal that's in React, it's pretty simple.